satpy/tests/reader_tests/test_ami_l1b.py

```diff
@@ -58,14 +58,25 @@ class TestAMIL1bNetCDFBase(unittest.TestCase):
     """Common setup for NC_ABI_L1B tests."""
 
     @mock.patch("satpy.readers.ami_l1b.xr")
-    def setUp(self, xr_, counts=None):
+    def setUp(self, xr_, counts=None, irtest=False):
```

Comment:
I didn't notice that `counts` was done this way too, but these kwargs to the setup function should be changed. I'd say either:

1. (Preferred) Separate out the data creation steps as a function that has to be called from within the tests (I haven't looked at how many tests depend on this) and pass the kwargs from there. You could also at that point switch this class to a pytest test (remove the unittest base class).
2. Same as 1, but make the data creation pytest fixtures. Not sure that provides much benefit given the keyword arguments.
3. Set `counts` and `irtest` as a class attribute and use `self.counts` and `self.irtest` inside the setup function.

Reply:
I have tried to update this, but some of the tests are now failing. Unfortunately I don't have more time to debug what's going on - I barely understand this testing stuff and it takes up too much time. So we'll have to leave it at this point. Am happy to revert to the old way (following what was in the tests before I made this PR).